### PR TITLE
feat(BILL-PCI14): make billing service reactive

### PR DESCRIPTION
**Pull request by ClementCadieux**
[JIRA: link to jira ticket](https://champlainsaintlambert.atlassian.net/jira/software/c/projects/PCI/boards/38/backlog?view=detail&selectedIssue=PCI-14&issueLimit=100)
## Context:
Converted Billing-Service to reactive
## Changes
Converted the entire service to webflux and used Mono and Flux
Removed mapper and made a EntityDTOUtil class with static method to handle that
Modified existing tests to conform to new code
Removed NotFoundException as it is not used
Modified error handling to be handled by EntityDTOUtil
## Before and After UI (Required for UI-impacting PRs)
No changes to UI
## Proofs
Runs on Docker
DockerRun
Builds on Gradle
GradleBuild
Jacoco Coverage
JacocoCoverage
Post from ApiGateway
PostmanSuccessPostFromApiGateway
Get from ApiGateway
PostmanSuccessGetFromApiGateway

### CODE REVIEW

**Review by Louenes099 — APPROVED** [review]
All looks good to me.
Would be nice to change Integers to Long for any object's ID.
The PR format (except the title missing a capital letter) is a model for everyone to follow

File: billing-service/src/main/java/com/petclinic/billing/datalayer/BillRepository.java [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
 import org.springframework.transaction.annotation.Transactional;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
 
-
-import java.util.List;
-import java.util.Optional;
-
-public interface BillRepository extends JpaRepository<Bill, Integer> {
+//Changed from JPA to ReactiveCRUD, makes the repo reactive without changing the entity
```
Comment (DuncanBH) [inline billing-service/src/main/java/com/petclinic/billing/datalayer/BillRepository.java:9]: Out of date comments?

**Review by DuncanBH — COMMENTED** [review]
Overall, pretty clean. Please break things down into smaller commits next time, very difficult to see the intent behind various changes here.

File: billing-service/src/main/java/com/petclinic/billing/datalayer/BillRepository.java (hunk 1 above)
Comment (ClementCadieux) [reply billing-service/src/main/java/com/petclinic/billing/datalayer/BillRepository.java:9]: Yup, ended up changing the repo to Mongo eventually

**Review by DuncanBH — APPROVED** [review]

**Review by AToner25 — APPROVED** [review]

**Pull request merged**